pipelines: Add KFP documentation for importing modelcars (#4097)
kubeflow/website: content/en/docs/components/pipelines/user-guides/components/importer-component.md

In addition to an `artifact_uri` argument, you must provide an `artifact_class` argument to specify the type of the artifact.

### Importing Model Artifacts from Container Images

Starting in Kubeflow Pipelines 2.5, you can import model artifacts that are packaged as container images using the [modelcar][model-car] format. Here's how it works:

1. **Specifying the URI**:

   - Use an OCI URI in the `artifact_uri` argument
   - Example: For a container image `quay.io/my-org/my-model:v1`, use `artifact_uri='oci://quay.io/my-org/my-model:v1'`

2. **Runtime Behavior**:

   - When a component uses the imported model, the modelcar runs as a sidecar container in the pod
   - The model's `path` property points to the `/models` directory in the running modelcar

3. **Handling Same User Requirements**:

   - The component container and modelcar container must run with the same user/UID

>    - If the user's don't match, accessing the model's `path` property will fail with a permission error

Reviewer: ```suggestion
   - If the users/UIDs don't match, accessing the model's `path` property will fail with a permission error
```

>    - No action is needed for Kubernetes distributions with random user assignment to pods such as OpenShift

Reviewer: ```suggestion
   - No action is needed for Kubernetes distributions with random user assignment to pods (such as OpenShift, for example)
```

   - For other cases, you have two options:
     - Build the modelcar container with the same UID as your component container image

>      - Set the `PIPELINE_RUN_AS_USER` environment variable to a nonroot UID on the Kubeflow Pipelines API server deployment to ensure consistent UIDs on all pods created by Kubeflow Pipelines

Reviewer: ```suggestion
     - Set the `PIPELINE_RUN_AS_USER` environment variable to a nonroot UID on the Kubeflow Pipelines API server deployment, which will ensure consistent UIDs on all pods created by Kubeflow Pipelines
```
